### ISSUE TITLE
API call /api/reload_config - result code != 0 when success

### ISSUE THREAD

**Issue by sarsonj** [processone/ejabberd#3319]
## Environment

- ejabberd version: 20.04

## Bug description

We are using API call reload_config to trigger ejabberd config reload. It works fine, however, the result code from this call is always 1. In ejabberd log seems to be everything file:

```
ejabberd-1 ejabberd-node 12:16:02.031 [info]  API call reload_config [] from ::ffff:10.244.0.202:37484
ejabberd-1 ejabberd-node
ejabberd-1 ejabberd-node 12:16:02.031 [info]  (#PID<0.2846.0>) Accepted connection [::ffff:10.244.0.202]:37484 -> [::ffff:10.244.0.27]:5444
ejabberd-1 ejabberd-node
ejabberd-1 ejabberd-node 12:16:02.031 [info]  Reloading configuration from config/ejabberd.yml
ejabberd-1 ejabberd-node
ejabberd-1 ejabberd-node 12:16:02.102 [info]  Configuration reloaded successfully
```

However, the result is 1:

```
root@ejabberd-0:/app# curl -d '{}' http://ejabberd-service:5444/api/reload_config
1
root@ejabberd-0:/app#
```

According to docs, it should be 0 when success: https://docs.ejabberd.im/developer/ejabberd-api/admin-api/

Could I ignore the result code or is there any problem with config reload?

**Comment by licaon-kter** [comment]
The reload might fail, test this by adding a `mod_bogus: {}` in modules, what happens then?

**Comment by sarsonj** [comment]
Hi, do I have to also setup something for mod_bogus? Because it fails now during start:
```
[error] Failed to set logging: {:error, {:not_found, :default}}
```

**Comment by licaon-kter** [comment]
What? Read again... that was just a bogus module to show you how it will fail on reload, of course it fails on start :) :)

**Comment by badlop** [comment]
I've checked using ejabberd master compiled from git, and the reload_config command returns 0 when the config hasn't changed, or when the changes are correct, both using mod_http_api and ejabberdctl:
```
$ ejabberdctl reload_config
$ echo $?
0
$ curl -d '{}' http://localhost:5443/api/reload_config
0
$
```

If I introduce a problem in the configuration file, the command is quite explicit about it:
```
$ curl -d '{}' http://localhost:5443/api/reload_config
"Invalid value of option modules: unknown ejabberd module: mod_bogus. Did you mean mod_bosh?"~ 
```

So, this raises some questions:
A) If you introduce a clear problem, like adding an inexistent module, what type of error does the command show to you?
B) If you keep the exact same configuration when starting and when reload_config, does it still return 1?
C) What does `ejabberdctl reload_config` return, 0 or 1?

**Comment by weiss** [comment]
This sounds like a dup of #3170, but that should be fixed in 20.04 ...
